fdbclient/ThreadSafeTransaction.cpp

```diff
@@ -541,16 +551,19 @@ Optional<TenantName> ThreadSafeTransaction::getTenant() {
 void ThreadSafeTransaction::operator=(ThreadSafeTransaction&& r) noexcept {
 	tr = r.tr;
 	r.tr = nullptr;
+	initialized = r.initialized;
+	r.initialized.reset();
 }
 
 ThreadSafeTransaction::ThreadSafeTransaction(ThreadSafeTransaction&& r) noexcept {
 	tr = r.tr;
 	r.tr = nullptr;
+	initialized = std::move(r.initialized);
```

Comment:
Should both the move constructors reset `r.initialized`?

Reply:
The move assignment should be leaving the source unset. From https://en.cppreference.com/w/cpp/memory/shared_ptr/operator%3D:

> 2) Move-assigns a shared_ptr from r. After the assignment, *this contains a copy of the previous state of r, and r is empty. Equivalent to shared_ptr<T>(std::move(r)).swap(*this).

It's probably good to write both implementations in the same way, and a move assignment that avoids changing the reference counts would be the cheaper alternative.